src/classes/DynamicMapOfCountry.cls

```diff
@@ -0,0 +1,167 @@
+        //this qurey working Parent(Country) to child(State) and child(District) to Parent(State) 
+        List<State__c>  listofState = [SELECT Id,country__r.Country_Code__c,country__r.Name,Name,
+                                            (SELECT state__r.Name,Name FROM Districts__r ) 
+                                        FROM State__c 
+                                        WHERE country__r.Country_Code__c = :countryCode];
+        for(State__c state :listofState) {
+            if(countryMap.size() == 0){
+                codeMap.put(state.country__r.Country_Code__c,new Map<String, Map<String, List<String>>>());
+                countryMap.put(state.country__r.Name,new Map<String, List<String>>());
+            }
+            stateMap.put(state.Name, new List<String>());
+            for(District__c district : state.Districts__r){
+                stateMap.get(district.state__r.Name).add(district.Name);
+            }
+        }
+        countryMap.get(listofState[0].country__r.Name).putAll(stateMap);
+        codeMap.get(listofState[0].country__r.Country_Code__c).putAll(countryMap);
+        
+        System.debug(codeMap.get(countryCode));
+        
+        List<String> listOfCombinations = new List<String>();
+        for(String countryName : countryMap.keySet()){
+            for(String stateName : countryMap.get(countryName).keySet()){
+               	for(String districtName : countryMap.get(countryName).get(stateName) ){
+                	if(countryName.startsWith(startLetter) && stateName.startswith(startLetter)
+                       && districtName.startsWith(startLetter) ){
+                        listOfCombinations.add(countryName+','+stateName+ ','+districtName);
+                    }
+             	}  	  
+            }
+        }
+        if(!listOfCombinations.isEmpty()){
+            System.debug('Combination of Country, State and District:'+listOfCombinations);
+        }
+        aggregateQurey();
+        countryData(name);
+    }
+
+    /**   Description :   Method Update the Field of Country by using Aggregate Qurey  
+    *     Created By  :   Sagar Nirwal
+    *     Arguments   :   none
+    *     Return      :   void
+    *     Revision log:   V_1.0 - Created  - Sagar Nirwal - 14/06/2022
+    *                     V_1.1 - Modified - Sagar Nirwal - 14/06/2022 - Aggregate Qurey
+    **/
+    public static void aggregateQurey(){
+
+        //Aggregate Qurey------> Total District Of State   
+        Map<Id, State__c> mapOfTotalDistricts = new Map<Id, State__c>();                                       
+        for(AggregateResult aggre : [SELECT  State__c state, COUNT(Name) 
+                                                      FROM District__c 
+                                                      GROUP BY State__c]){
+            mapOfTotalDistricts.put(String.valueOf(aggre.get('state')),new State__c(Id = (Id)aggre.get('state'),
+                Total_District__c = (Decimal)aggre.get('expr0')));
```

Comment:
Replace state with State__c and expr0 with totalDist while getting value.
And do same changes in below query as well.